### PR TITLE
Bugfixing small issues related to title links

### LINKED ISSUE #1763: The user can still edit the default source by clicking on the title
### Describe the bug

Although the dropdown is hidden for the default source, a user can still go to the edit screen if she clicks on the source name

### Steps to reproduce

1. Go to 'LogAnalysis' > 'Sources'
2. Click on 'panther-account-...' title on the first source
3. You can edit the default source

### Expected behavior

The default source should not be editable

### LINKED ISSUE #1829: Log Analysis Overview: Alert title clickable area extends across the entire page
### Describe the bug

In the Recent Alerts and High Severity Alerts tables, the alert title is clickable. This takes you directly to the alert. However, the clickable area of the alert title extends across the entire page. This makes it very easy to accidentally click through to the alert when scrolling the list or trying to select the triage options.

### Steps to reproduce

1. Go to the Log Analysis Overview page
2. Scroll down to the Recent Alerts section
3. Hover your mouse over the alert title, the cursor should change to indicate it is clickable
4. Slide your mouse cursor all the way to the right of the page, the clickable cursor should stay
5. Click on the far right side of the page, you'll get redirected to the alert details page

### Expected behavior

Only the actual title text is clickable.

### Environment

How are you deploying or using Panther?

- Panther version 1.11 release candidate

### Screenshots

Screen Shot 2020-10-21 at 9 48 19 AM

### LINKED ISSUE #1892: Log Analysis Rules - Timestamp set incorrectly
### Describe the bug

On the Log Analysis Rules page, the timestamps being displayed in the chart are incorrect. They appear to always be the current time.

### Steps to reproduce

1. Go to Log Analysis Rules
1. Page through rules. They will all show the same Time Created timestamp.
1. Click into a rule. The Last Modified and Time Created timestamps will differ from the Time Created timestamp on the overview page (most likely, I guess it's possible if you just created the rule that minute they are the same).

### Expected behavior

The rules overview page correctly displays the Last Modified time.

### Environment

How are you deploying or using Panther?

- Panther version v1.11.0 from source deployment.

### Additional context

It appears that the timestamp displayed is always just the current time, which is incorrect.

The label is Time Created, but graphql is returning Last Modified.

I verified that the correct values are stored in the backend in DynamoDB.

Time Created is not so useful as a metric here, we really want Last Modified.

### Screenshots

Screen Shot 2020-10-29 at 10 34 08 AM

### LINKED ISSUE #1894: Log Analysis Rules - UX - Allow text entry in Log Type box
### Describe the ideal solution

On the Log Analysis Rules page you can filter the results by log type. You do this by clicking the Log Types drop down in the top right of the page. When you click on this box, you can scroll through all the available log types and click on one to select it.

You are not able to enter text into this box to filter to the log type you want. This makes navigating to a log type near the bottom or middle of the list a hassle. Ideally I would be able to start typing the log type  and the dropdown would filter to just log types that match what I'm typing.

In the past I have used the `bug` label for UX improvement requests, but that seems incorrect. I'm trying out the `enhancement` label here, that seems more appropriate.  Please let me know how the web team feels about this 🙏 

### Describe your use cases

This would speed up my ability to filter to specific log types that I care about.

### How frequently would you use such feature

Weekly.

### Describe alternatives you have considered

Displaying all the log types at once might work. Or alternatively only displaying enabled log types would make the list much smaller and more useful (I don't know if it is currently possible for the frontend to query for this information).

### Additional context

Screen Shot 2020-10-29 at 10 55 33 AM

### PR DESCRIPTION
**Pull request by alexmylonas**
## Description

Small PR fixing multiple of small bugs: 

- [Log Analysis Overview: Alert title clickable area extends across the entire page](https://github.com/panther-labs/panther/issues/1829)
- [Removed link from title on SourceCard on panther created log sources](https://github.com/panther-labs/panther/issues/1763)

- [Log Analysis Rules - UX - Allow text entry in Log Type box](https://github.com/panther-labs/panther/issues/1894)
- [Log Analysis Rules - Timestamp set incorrectly](https://github.com/panther-labs/panther/issues/1892
)

Closes #1829, closes #1763, closes #1894, closes #1892 

## Tests 

- Locally

### CODE REVIEW

File: web/src/components/cards/AlertCard/AlertCard.tsx [hunk 1]
```diff
@@ -44,14 +44,16 @@ const AlertCard: React.FC<AlertCardProps> = ({ alert, hideRuleButton = false })
...
-        <Link
-          as={RRLink}
-          aria-label="Link to Alert"
-          to={urls.logAnalysis.alerts.details(alert.alertId)}
-          cursor="pointer"
-        >
-          <GenericItemCard.Heading>{alert.title}</GenericItemCard.Heading>
-        </Link>
+        <GenericItemCard.ValuesGroup>
+          <Link
```
Comment (3nvi) [inline web/src/components/cards/AlertCard/AlertCard.tsx:48]: This fixes the issue at the `AlertCard` level, but this is a `GenericItemCard` issue. I'd suggest to simply add `truncated` prop on the `GenericItemCardHeading` component.

While there, you could also remove `cursor: pointer` from all Links that are wrapping a `GenericItemCard.Heading` component cause I believe it's not needed

Comment (alexmylonas) [reply web/src/components/cards/AlertCard/AlertCard.tsx:48]: Removed `cursor: pointer`. 

Revert my change and changed every instance of `GenericItemCard.Heading`:  
```
        <Link
          as={RRLink}
          aria-label="Link to Alert"
          to={urls.logAnalysis.alerts.details(alert.alertId)}
          cursor="pointer"
        >
          <GenericItemCard.Heading>{alert.title}</GenericItemCard.Heading>
        </Link>
```
To: 
```
          <GenericItemCard.Heading>
            <Link
              as={RRLink}
              aria-label="Link to Rule"
              to={urls.logAnalysis.rules.details(rule.id)}
            >
              {rule.displayName || rule.id}
            </Link>
          </GenericItemCard.Heading>
```

**Comment by alexmylonas** [comment]
Addressed feedback, check above!

Also added fixes for:
- [Log Analysis Rules - UX - Allow text entry in Log Type box](https://github.com/panther-labs/panther/issues/1894)
- [Log Analysis Rules - Timestamp set incorrectly](https://github.com/panther-labs/panther/issues/1892
)

File: web/src/pages/ListDestinations/DestinationCards/DestinationCard.tsx [hunk 2]
```diff
@@ -37,13 +37,15 @@ const DestinationCard: React.FC<DestinationCardProps> = ({ destination, logo, ch
...
-          to={urls.settings.destinations.edit(destination.outputId)}
-          cursor="pointer"
-        >
-          <GenericItemCard.Heading>{destination.displayName}</GenericItemCard.Heading>
-        </Link>
+        <GenericItemCard.Heading>
+          <Link
+            as={RRLink}
+            to={urls.settings.destinations.edit(destination.outputId)}
+            cursor="pointer"
```
Comment (3nvi) [inline web/src/pages/ListDestinations/DestinationCards/DestinationCard.tsx:44]:
```suggestion

```

File: web/src/components/cards/RuleCard/RuleCard.tsx [hunk 3]
```diff
@@ -66,7 +67,10 @@ const RuleCard: React.FC<RuleCardProps> = ({ rule }) => {
           <GenericItemCard.ValuesGroup>
             <Flex ml="auto" mr={0} align="flex-end" spacing={4}>
               <GenericItemCard.Value label="Threshold" value={formatNumber(rule.threshold)} />
-              <GenericItemCard.Value label="Time Created" value={formatDatetime(rule.createdAt)} />
+              <GenericItemCard.Value
```
Comment (3nvi) [inline web/src/components/cards/RuleCard/RuleCard.tsx:70]: why are we changing it to `rule. lastModified`?

Comment (alexmylonas) [reply web/src/components/cards/RuleCard/RuleCard.tsx:70]: According to this [issue](https://github.com/panther-labs/panther/issues/1892): 

> Expected behavior
> The rules overview page correctly displays the Last Modified time.

Comment (3nvi) [reply web/src/components/cards/RuleCard/RuleCard.tsx:70]: ah gotcha. Missed that.

File: web/src/pages/ListDestinations/DestinationCards/DestinationCard.tsx (hunk 2 above)
Comment (alexmylonas) [reply web/src/pages/ListDestinations/DestinationCards/DestinationCard.tsx]: Removed!

File: web/src/pages/ListLogSources/LogSourceCards/LogSourceCard.tsx [hunk 4]
```diff
@@ -67,13 +67,17 @@ const LogSourceCard: React.FC<LogSourceCardProps> = ({ source, children, logo })
...
-        </Link>
+        {!isCreatedByPanther ? (
+          <GenericItemCard.Heading>
+            <Link as={RRLink} to={urls.logAnalysis.sources.edit(source.integrationId, sourceType)}>
+              {source.integrationLabel}
+            </Link>
+          </GenericItemCard.Heading>
+        ) : (
+          <GenericItemCard.Heading>
+            <Text color="blue-200">{source.integrationLabel}</Text>
```
Comment (3nvi) [inline web/src/pages/ListLogSources/LogSourceCards/LogSourceCard.tsx:78]: This is not a link anymore, so IMO it shouldn't have the color of a link in order not to confuse the user.

I propose a `teal-300` that will make it "stand out" as differentt, but not mark it as link
```suggestion
            <Text color="teal-300">{source.integrationLabel}</Text>
```

**Review by 3nvi — APPROVED** [review]
Looks good @alexmylonas . Just a minor comment

**Pull request merged**